common/Timer.cc

```diff
@@ -47,16 +47,17 @@ void Timer::cancel() {
     this->canceled = true;
 }
 
-Timer Timer::clone() const {
+unique_ptr<Timer> Timer::clone() const {
```

Comment:
Making this return a `unique_ptr` seems wrong to me: you can `move` it in on the use-site.
Could we do that instead?

Reply:
For some reason, it was triggering the copy constructor instead of the move constructor.

I can see if I can make it work.